### PR TITLE
VI: Change force-progressive to be more compatible

### PR DESCRIPTION
**Pull request by booto**
Trying to address https://bugs.dolphin-emu.org/issues/9023
Needs testing @JMC47

### CODE REVIEW

File: Source/Core/Core/HW/VideoInterface.cpp [hunk 1]
```diff
@@ -558,24 +567,12 @@ static void BeginField(FieldType field)
...
-		}
-	}
-	else
-	{
-		if (field == FieldType::FIELD_EVEN)
-		{
-			xfbAddr = GetXFBAddressTop();
+
+		if ((field == FieldType::FIELD_ODD) && (m_VBlankTimingOdd.PRB == m_VBlankTimingEven.PRB + 1)) {
+			xfbAddr -= (fbStride*2);
```
Comment (neobrain) [inline Source/Core/Core/HW/VideoInterface.cpp]: This trades declarative programming style (the code explicitly says what it's doing) against what's undocumented black magic (or maybe it is documented somewhere in the block above, either way it's much less apparent what it's doing).

Is there any chance you can document these lines, or even better - use a declarative code flow here?

**Comment by phire** [comment]
This looks a lot like the old code, which broke for a whole bunch of PAL games.

**Comment by phire** [comment]
Nevermind. This code looks better on reading it a second time. I also tested this with a bunch of PAL games and found no issues.

**Comment by neobrain** [comment]
@booto why?

**Comment by booto** [comment]
This PR was mostly used to generate a build for testing. It didn't fix the problem is was meant to address. A different PR (#3252) was created with what I think is a more correct fix.

**Pull request closed without merging**